asllib/doc/ASLSemanticsReference.tex

```diff
@@ -291,107 +2073,73 @@ \section{SemanticsRule.EGlobalVar \label{sec:SemanticsRule.EGlobalVar}}
     the evaluation of~\texttt{global\_x} within~\texttt{assert global\_x == 3;}
     uses the rule SemanticsRule.EGlobalVar.
 
-%  \subsection{Example: SemanticsRuleEGlobalVarError.asl}
-%    The program:
-%    \VerbatimInput{../tests/ASLSemanticsReference.t/SemanticsRule.EGlobalVarError.asl}
-%    raises a type error because \texttt{x} is already declared locally in
-%    \texttt{stdlib.asl}.
-
   \subsection{Code}
   \VerbatimInput[firstline=\EGlobalVarBegin, lastline=\EGlobalVarEnd]{../Interpreter.ml}
 
 \begin{formal}
-  \subsection{Formally: sequential case}
-  \begin{align}
-  \interp{\texttt{x}}(\texttt{env}) & \triangleq \left\{ (\texttt{env}[\texttt{x}], \texttt{env})) \right\}
-  \label{eq:sem-seq-var}
-  \end{align}
-
-  \subsection{Formally: concurrent case}
-  \begin{align}
-  \interp{\texttt{x}}(\texttt{env}) & \triangleq \left\{ (\texttt{env}[\texttt{x}], \texttt{env}, \texttt{R(x)}) \right\}
-  \label{eq:sem-conc-var}
-  \end{align}
+  \subsection{Formally}
+Due to how $\denv[\cdot]$ distinguishes between local variables and global variables,
+we can use the same rule as for local variable expressions:
```

Comment:
why do we have two rules?

Reply:
I removed the denv[.] definitions so now the two rules are different and clearer.